### ISSUE TITLE
local storage and Hetzner storage share

### ISSUE THREAD

**Issue by guerby** [nextcloud/backup#279]
Hi,

I've tried the backup app on a nextcloud managed by hetzner:

https://www.hetzner.com/storage/storage-share

The app puts the nextcloud in maintenance mode as expected but when it gets out of maintenance nothing is visible and backup page says "in the next 5 minutes".

In the backup app settings the "local storage" path is greyed out and not editable.

Since hetzner nextcloud doesn't allow external storage I think the current backup app cannot work there as is

May be if the backup app configuration would allow to specify a place in the nextcloud user visible directory (with of course implicit ".nobackup" for this directory to avoid rebacking up infinitely) it might be usable in hetzner managed nextcloud context.

May be just a checkbox and hardcoded "backup-app" directory for the admin account of the instance?

**Comment by AnAnalogGuy** [comment]
I'm using https://www.hetzner.com/storage/storage-box as external storage for a nextcloud installation connected via SFTP. So it basically works. 
An initial full snapshot might take hours, so maybe what you see is how it looks like if the backup is in progress.

**Comment by karlemilnikka** [comment]
@AnAnalogGuy, That’s probably a good solution for many Nextcloud users. It is unfortunately not a viable solution for Hetzner’s Storage Share users since they can’t connect Storage Boxes over SFTP as external storage.

**Comment by AnAnalogGuy** [comment]
But why not just using a Hetzer Storage Box instead of a Hetzner Storage Share?

**Comment by karlemilnikka** [comment]
You can’t run Nextcloud on a Storage Box.

**Comment by karlemilnikka** [comment]
Yes. That’s how I understood the issue’s background. It’s indeed an issue with Hetzner’s Storage Share service and not a Nextcloud issue. It would however be practical to have a local backup alternative, so I think it’s a good feature suggestion.

**Comment by guerby** [comment]
> Ah, I see, sorry, i didn't get what @guerby actually wanted to achive - using the backup tool from the Hetzner NC installation on Storage Share to back up to a third party storage. Sorry, my bad.
> 
> I guess he might have to address his request to Hetzner then, as it's not a NC issue.

It's indeed an Hetzner offer limitation but I think most of the pieces are in place in NC backup to make it work even on Hetzner which does not offer a direct nextcloud access. 

Being able to generate a backup file inside a user choosen visible nextcloud directory (with automatic .nobackup in there) might have uses outside Hetzner as well, for example to extract backups using nextcloud API and not ssh commands, share backups easily, etc...

**Comment by blunix0815** [comment]
you can use the Hetzner Storage Box for the data directory directly. I am currently also in the process of setting up a nextcloud on Hetzner Cloud and found [this](https://www.blunix.com/blog/howto-install-nextcloud-on-ubuntu-2204-with-hetzner.html) blogpost describes everything i needed.

**Comment by guerby** [comment]
> you can use the Hetzner Storage Box for the data directory directly. I am currently also in the process of setting up a nextcloud on Hetzner Cloud and found [this](https://www.blunix.com/blog/howto-install-nextcloud-on-ubuntu-2204-with-hetzner.html) blogpost describes everything i needed.

Thanks for the link, it is indeed possible when using a dedicated server to use a storage box as storage backend. 

However in cases you don't have root access as in a storage share it would still be nice to be able to do backups.
